Overhauled Angle-based Camera system

AL2009man opened the pull request:
~~This is a two-part PR, the Overhauled Camera System, and the Decoupled Crosshair Mouse sensitivity. Originally planned to make it separate, but due to issues specific to Crosshair Camera, this PR merged both into one.~~

----

To prepare for Flick Stick support: I went ahead and basically overhaul the entire camera system to be based on 360-degree rotational/angle system, this should improve general camera controlling system when using Mouse and be more friendly towards custom Reset Camera macros like SteamInput.  It also prepares for future Input devices like Gyro Aiming (#644) and Flick Stick

This also features improvements to Crosshair Aiming, to make it more preciser-- even closer to raw movement!

AL2009man committed:
Overhauled Camera system + Decoupled Mouse Crosshair

This PR overhauls the entire Camera system to better allow for future input methods like Gyro and Flick Stick, alongside decoupling Mouse Crosshair Speed out of Mouse Camera Speed, making the Crosshair Mouse be independent.

AL2009man committed:
Normalized Mouse Crosshair Swivel

to equalize the swiveling between Horts/Verts crosshair swivel for mouse input: the swivel has been reduced to 20%

fgsfdsfgs commented on port/src/input.c:
```diff
@@ -1277,6 +1278,29 @@ void inputMouseSetSpeed(f32 x, f32 y)
+	g_PlayerExtCfg[0].mouseaimsensx = x;
+	g_PlayerExtCfg[0].mouseaimsensy = y;
+}
+
```
If possible, don't reference game code stuff here. Doesn't look like these functions have to be in input.c either, since you can just get the unscaled mouse delta in the bmove code and work with it the same way.

AL2009man replied:
hit a bit of a snag while fixing HoverBike and CamSpy. will have to rollback that for now 

but it'll be in a new test branch, head to https://github.com/AL2009man/perfect_dark/tree/CentralizedCameraSystem-test

AL2009man replied:
ok, this should be resolved properly

AL2009man committed:
moved `inputMouseGetScaledDeltaCrosshair` to `bondmove.c` directly

moved the input-specific layer over to bmove directly, this means the original function has been removed

AL2009man committed:
Merge branch 'fgsfdsfgs:port' into PerfectQuakeDark3

AL2009man committed:
fixed regression on CamSpy and Hoverbike

moved the original scaling from bmove over to input.c, while addressing issues with CamSpy and Hoverbike sensitivity. the latter now has customized scaling solution specific

the `inputMouseGetScaledDeltaCrosshair` has been restored due to a specific bug with overall camera scaling after fixing CamSpy and Hoverbike, but itt'll be temporary, depending on the next commit fix?

AL2009man committed:
re-move MouseGetScaledDeltaCrosshair scaling back to bmove

redid the migration, now it should work without any conflictions

AL2009man commented:
as requested: I already went thru a initial phase of testing. Right now: there should be no more issues left with the Overhauled Camera system that broke SpyCam and Hoverbike. As a reminder: the overhauled camera system only applies to Mouse Camera (since it's used as a basis), Classic Joystick should remained the same as before.

however: a few things that might need advanced testings, the first two is where i can't directly test it myself.

- [ ] Port-net
- [ ] demo
- [ ] in-game cutscene (most notable with Mission 1 Cour 3)

AL2009man committed:
fixed regression in Crosshair Swivel.

revised the entire crosshair swivel system to be closer to the original setup. This also fixes a problem with the crosshair not working as intended.

AL2009man committed:
fixed regression on Crosshair Sway while using Mixed Input and clamped swaying at high sensitivity

fixed a regression where the clamped swaying was missing, while restoring mixed input-based crosshair sway

AL2009man committed:
cleaned up AngleCamera, attempt to address netplay/demo compatibility

cleaned up AngleCamera to make it a bit easier to manage and easily add or remove Camera-related inputs.

also, a experimental fix to netplay/demo compatibility-- but it's hasn't really been tested outside of initial gameplay-based cutscenes

AL2009man committed:
fixed reverse whitespace

AL2009man committed:
Merge branch 'fgsfdsfgs:port' into PerfectQuakeDark3

AL2009man committed:
fixed regression with Mouse Crosshair Aiming always being active regardless of Joystick Crosshair state

Noticed some parts of the code was removed by accident, but that should be restored to how it originally was.

Additionally: inorder to retain the sensitivity-tied fix: movedata.freelookdx/y has been replaced with `inputMouseGetRawDelta`'s equiavlent, although: that one's a workaround for the time being

AL2009man committed:
remove decoupled crosshair infavor with `mouseaimspeedx/y`

Inorder to make PR reviewing easier, the crosshair decouple has been removed. To compensate: there's now a internal negative multiplayer for crosshair to compensate

AL2009man committed:
Merge branch 'port' into PerfectQuakeDark3

AL2009man committed:
replaced temp movedata with actual camera yaw/pitch

AL2009man committed:
Revert "replaced temp movedata with actual camera yaw/pitch"

This reverts commit 2348fff3a8aeacd465b40f18b177a214b6a178ee.

AL2009man committed:
Reworked Angle-based Camera system to directly use the engine camera angle

unlike the previous method: this one directly use the engine's angle camera. (this is also a squash merge test)

AL2009man committed:
Merge branch 'port' into PerfectQuakeDark3

AL2009man committed:
Fix missing newline at end of `optionsmenu.c`
